Automattic/jetpack
# Search: Add Search Dashboard support for WPCOM simple sites

#### Changes proposed in this Pull Request:
The PR adds Search Dashboard support for WPCOM simple sites, including

- Extracts rest controller name space to a static variable
- Splits routes in three groups, Jetpack only, WPCOM only and common
- Extracts priority to add search menu to a variable - on WPCOM we need different priorities
- Extracted some functions to be override
- Treat WPCOM sites as always connected

#### Other information:


#### Jetpack product discussion
pcNPJE-UW-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Please following instructions in `D84235-code`
- Ensure Search dashboard still works for JP sites
